src/m365/commands/login.ts

```diff
@@ -199,32 +203,36 @@ class LoginCommand extends Command {
         auth.connection.cloudType = CloudType.Public;
       }
 
-      try {
-        await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);
-        auth.connection.active = true;
-      }
-      catch (error: any) {
-        if (this.debug) {
-          await logger.logToStderr('Error:');
-          await logger.logToStderr(error);
-          await logger.logToStderr('');
-        }
-
-        throw new CommandError(error.message);
-      }
+      await this.obtainAccessToken(logger);
 
       const details = auth.getConnectionDetails(auth.connection);
 
       if (this.debug) {
         (details as any).accessToken = JSON.stringify(auth.connection.accessTokens, null, 2);
       }
 
-
       await logger.log(details);
     };
 
-    deactivate();
-    await login();
+    try {
+      if (this.shouldRenewConnection(args.options)) {
+        await auth.clearConnectionInfo();
+      }
+    }
+    catch (error: any) {
+      if (this.debug) {
+        await logger.logToStderr(new CommandError(error));
+      }
+    }
+    finally {
+      if (!this.shouldRenewConnection(args.options)) {
+        await this.obtainAccessToken(logger);
```

Comment:
Since we already refreshed access token on line 206, what's the idea behind repeating it here?

Reply:
In line 206 the accessToken is retrieved, as part of the entire login functionality. Here on line 229 we only want to refresh the accessToken and not go through the entire login functionality. Additionally, if we had to call the login functionality here, we would end up with `auth.getConnectionDetails(auth.connection)` fail because when refreshing the access token the `connection.name` is undefined.

Please let me know if you have another vision to this so we can align